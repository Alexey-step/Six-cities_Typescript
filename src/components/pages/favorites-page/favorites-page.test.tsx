import * as React from "react";
import {render, screen} from "@testing-library/react";
import FavoritesPage from "./favorites-page";
import * as redux from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from "react-router-dom";
import {createMemoryHistory} from "history";
import {offers, authInfo} from "../../../mocks/tests-mocks";

const mockStore = configureStore([]);
describe(`Test FavoritesPage component`, () => {
  const history = createMemoryHistory();
  it(`FavoritesPage component should render correctly`, () => {
    const store = mockStore({
      MAIN: {isError: false},
      USER: {authInfo: authInfo.adapted},
      FAVORITE: {favorites: offers.adapted}
    });
    store.dispatch = jest.fn();
    render(
        <redux.Provider store={store}>
          <Router history={history}>
            <FavoritesPage />
          </Router>
        </redux.Provider>
    );
    expect(screen.getByText(/Saved listing/!)).toBeInTheDocument();
  });
  it(`FavoritesPage component should render correctly when favorites empty`, () => {
    const store = mockStore({
      MAIN: {isError: false},
      USER: {authInfo: authInfo.adapted},
      FAVORITE: {favorites: []}
    });
    store.dispatch = jest.fn();
    render(
        <redux.Provider store={store}>
          <Router history={history}>
            <FavoritesPage />
          </Router>
        </redux.Provider>
    );
    expect(screen.getByText(/Save properties to narrow down search or plan your future trips./!)).toMatchSnapshot();
  });
});
