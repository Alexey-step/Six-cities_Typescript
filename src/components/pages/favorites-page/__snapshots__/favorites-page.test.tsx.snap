// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FavoritesPage component FavoritesPage component should render correctly when favorites empty 1`] = `
<p
  class="favorites__status-description"
>
  Save properties to narrow down search or plan your future trips.
</p>
`;
